src/qos_core/src/protocol/services/key.rs

```diff
@@ -59,6 +59,13 @@ pub(in crate::protocol) fn inject_key(
 	// 4. Write the Quorum Key to the file system, at which point New Node will
 	// automatically pivot to running the Pivot App.
```

Comment:
What is this "automatically pivot"? Does that imply a race condition with step 5 below?
i.e. could the pivot happen really fast and get the old (unrotated) key?

Reply:
At first I thought this comment was incorrect or misleading, but indeed: we do rely on "quorum key file exists" as a signal to start the pivot app:
https://github.com/tkhq/qos/blob/6ccf4e2fe457a00018a520b6d6f3f7c483b0dc01/src/qos_core/src/reaper.rs#L56-L67

So there's a potential race condition indeed: if the app is started in between the quorum key write and the ephemeral key rotation, the app will read the old ephemeral key. However unlikely this is, we should address it, and it's easy enough to fix: rotate the ephemeral key _first_, then write the quorum key.

Also...I'll update the comment to be more specific.

Great catch!